src/input.rs

```diff
@@ -569,6 +566,24 @@ mod tests {
         assert_eq!(err.to_string().contains(filename), true);
     }
 
+    #[test]
+    fn test_input_gzip() {
+        use std::io::Cursor;
+        let data = Cursor::new(
+            // echo '' | gzip -cf | od -A n -t u1 -w40| sed -e 's/^  *//; s/  */, /g'
+            vec![31, 139, 8, 0, 0, 0, 0, 0, 0, 3, 227, 2, 0, 147, 6, 215, 50, 1, 0, 0, 0],
+        );
+        let mut input = Input::open_stream(&PathBuf::from("test.log.gz"), Box::new(data)).unwrap();
+        let mut buf = [0; 32];
```

Comment:
Please use stricter check here
```rust
        let mut buf = Vec::new();
        let result = input.stream.read_to_end(&mut buf);
        assert!(result.is_ok());
        assert_eq!(result.unwrap(), 1);
        assert_eq!(buf, vec![0x0a]);
```

Reply:
You can also add some text to echo and check that it matches in the end.
Like 
```rust
        let data = Cursor::new(
            // echo 'test' | gzip -cf | god -A n -t u1 -w40| sed -e 's/^  *//; s/  */, /g'
            vec![
                31, 139, 8, 0, 158, 217, 72, 103, 0, 3, 43, 73, 45, 46, 225, 2, 0, 198, 53, 185, 59, 5, 0, 0, 0,
            ],
        );
        // ...
        assert_eq!(buf, b"test\n");
```

Reply:
Also, the shell command and the input binary data literal can be further simplified like this
```rust
        let data = Cursor::new(
            // echo 'test' | gzip -cf | xxd -p | sed 's/\(..\)/\\x\1/g'
            b"\x1f\x8b\x08\x00\x9e\xdd\x48\x67\x00\x03\x2b\x49\x2d\x2e\xe1\x02\x00\xc6\x35\xb9\x3b\x05\x00\x00\x00",
        );
```